plugins/pkg/file/fs_test.go

```diff
@@ -16,28 +16,74 @@ package file
 
 import (
 	"os"
+	"path/filepath"
+	"sync"
 	"testing"
-	"time"
 
+	"github.com/fsnotify/fsnotify"
 	"github.com/stretchr/testify/assert"
 )
 
-func TestFileMtimeDetection(t *testing.T) {
-	defaultFs = newFS(2000 * time.Millisecond)
+func TestFileIsChanged(t *testing.T) {
+	var wg sync.WaitGroup
+	var mu sync.Mutex
+	i := 4
 
-	tmpfile, _ := os.CreateTemp("", "example")
-	defer os.Remove(tmpfile.Name()) // clean up
+	tmpfile, _ := os.CreateTemp("./", "example")
+	defer func(name string) {
+		err := os.Remove(name)
+		if err != nil {
+			t.Logf("%v", err)
+		}
+	}(tmpfile.Name())
+	file, err := Stat(tmpfile.Name())
 
-	f, err := Stat(tmpfile.Name())
+	assert.NoError(t, err)
+	assert.Equal(t, tmpfile.Name(), file.Name)
+	err = WatchFiles(func() {
+		wg.Add(1)
+		defer wg.Done()
+		mu.Lock()
+		i = 5
```

Comment:
Using a boolean is enough for this test. And as the var is only updated in the callback, and access after wg.Wait, we don't need a lock to protect it.